src/providers/oauth-provider.js

```diff
@@ -71,6 +49,37 @@ function OAuthProvider() {
     }
 
     return config;
+}
+
+
+/**
+ * OAuth provider.
+ */
+
+function OAuthProvider() {
+  var defaultConfig;
+
+  /**
+   * Configure.
+   *
+   * @param {object} params - An `object` of params to extend.
+   */
+
+  this.configure = function(params) {
+    // Can only be configured once.
+    if (defaultConfig) {
+      throw new Error('Already configured.');
+    }
+
+    // Check if is an `object`.
+    if (!(params instanceof Object)) {
+      throw new TypeError('Invalid argument: `config` must be an `Object`.');
+    }
+
+    // Extend default configuration.
+    defaultConfig = angular.extend({}, defaults, params);
```

Comment:
Instead of doing this, I think i'll be simpler to add a new argument to each method called `config`. Then on each method, you just need to extend `defaultConfig` with the those values.